File: 1395.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution {
public:
    int numTeams(vector<int>& rating) {
        int n = rating.size();
        int ans = 0;
        for(int i = 0; i < n; i++){
            int left_less = 0, left_more = 0, right_less = 0, right_more = 0;
            for(int j = 0; j < i; j++){
                if(rating[j] < rating[i]) left_less++;
                else if(rating[j] > rating[i]) left_more++;
            }
            for(int k = i + 1; k < n; k++){
                if(rating[k] < rating[i]) right_less++;
                else if(rating[k] > rating[i]) right_more++;
            }
            ans += left_less * right_more + left_more * right_less;
        }
        return ans;
    }
};

int main(){
    int n;
    cin >> n;
    vector<int> rating(n);
    for(int i = 0; i < n; i++)
        cin >> rating[i];
    Solution s;
    cout << s.numTeams(rating) << endl;
    return 0;
}

File: 1395.py
class Solution:
    def numTeams(self, rating: list[int]) -> int:
        n = len(rating)
        ans = 0
        for i in range(n):
            left_less = left_more = right_less = right_more = 0
            for j in range(i):
                if rating[j] < rating[i]:
                    left_less += 1
                elif rating[j] > rating[i]:
                    left_more += 1
            for k in range(i + 1, n):
                if rating[k] < rating[i]:
                    right_less += 1
                elif rating[k] > rating[i]:
                    right_more += 1
            ans += left_less * right_more + left_more * right_less
        return ans
    
if __name__ == '__main__':
    rating = list(map(int, input().split()))
    print(Solution().numTeams(rating))